solr/core/src/java/org/apache/solr/search/QueryUtils.java

```diff
@@ -83,6 +85,25 @@ public static boolean isConstantScoreQuery(Query q) {
     }
   }
 
+  public static void ensurePrefixQueryObeysMinimumPrefixLength(
+      Query query, String prefix, int minPrefixLength) {
+    // TODO Should we provide a query-param to disable the limit on a request-by-request basis?  I
```

Comment:
yes indeed; in fact the QParser default algorithm could lookup a local-param (not request level) like prefixQueryMinimumLength.  This will be straightforward once you create the QParser default method as I suggested.